### PR TITLE
unnecessary params during get_or_create warning

### PR DESCRIPTION
**Pull request by evgeniilezhnin**

### CODE REVIEW

File: client/verta/verta/client.py [hunk 1]
```diff
@@ -627,7 +654,7 @@ def set_dataset(self, name=None, type="local",
 
         If an accessible Dataset with name `name` does not already exist, it will be created
         and initialized with specified metadata parameters. If such a Dataset does already exist,
-        it will be retrieved; specifying metadata parameters in this case will raise an exception.
+        it will be retrieved; specifying metadata parameters in this case will raise a warning.
```
Comment (nhatsmrt) [inline client/verta/verta/client.py:657]: Hmm where is this enforced? I can't seem to find the use of check_unnecessary_params_warning for set_dataset.

**Review by nhatsmrt — COMMENTED** [review]
This might seem tedious, but can you add a small test to verify that the warnings is thrown correctly? Maybe write a test for something that can be created quickly (e.g Project), or tag this along in one of the existing test_create_*'s. pytest has this construct that should allow you to check if a warning is thrown:

https://docs.pytest.org/en/stable/warnings.html#warns

File: client/verta/verta/client.py (hunk 1 above)
Comment (evgeniilezhnin) [reply client/verta/verta/client.py:657]: It was there before the current PR here: https://github.com/VertaAI/modeldb/blob/master/client/verta/verta/_dataset.py#L60

**Comment by evgeniilezhnin** [comment]
> This might seem tedious, but can you add a small test to verify that the warnings is thrown correctly? Maybe write a test for something that can be created quickly (e.g Project), or tag this along in one of the existing test_create_*'s. pytest has this construct that should allow you to check if a warning is thrown:
> 
> https://docs.pytest.org/en/stable/warnings.html#warns

good idea, I've added test cases

File: client/verta/verta/client.py (hunk 1 above)
Comment (nhatsmrt) [reply client/verta/verta/client.py:657]: Oh I missed that. Nice.

**Review by nhatsmrt — APPROVED** [review]
LGTM, assuming tests passed.

File: client/verta/tests/test_dataset_versioning/test_dataset.py [hunk 2]
```diff
@@ -31,6 +31,8 @@ def test_create(self, client, created_datasets):
         excinfo_value = str(excinfo.value).strip()
         assert "409" in excinfo_value
         assert "already exists" in excinfo_value
+        with pytest.warns(UserWarning, match='.*already exists.*'):
```
Comment (convoliution) [inline client/verta/tests/test_dataset_versioning/test_dataset.py:34]: Oh wow, I didn't know there  is a `match` parameter. That's useful!

**Pull request merged**